File: Data Structures & Algorithms/jump-game-ii/submission-3.cpp
class Solution {
public:
    unordered_map<int,int>ump;
    int go(vector<int>&nums,int idx){
        if(idx==nums.size()-1) return 0;
        if(nums[idx]==0) return INT_MAX-100;
        if(ump.find(idx)!=ump.end()) return ump[idx];
        int ans = INT_MAX;
        int n = min((int)nums.size()-1, idx+nums[idx]);
        for(int i=idx+1;i<=n;++i){
            ans = min(ans, 1+go(nums,i));
        }
        ump[idx]= ans;
        return ump[idx];
    }
    int jump(vector<int>& nums) {
        return go(nums,0);
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
import copy
class Solution:
    ans = []
    def go(self,nums, temp_nums,idx, target):
        if target==0:
            self.ans.append(copy.copy(temp_nums))
            return 
        if target<0 or idx==len(nums):
            return 
        temp_nums.append(nums[idx])
        target-=nums[idx]
        self.go(nums,temp_nums,idx,target)

        target+= nums[idx]
        temp_nums.pop()
        self.go(nums,temp_nums,idx+1,target)
        
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        self.ans = []
        self.go(nums,[],0,target)
        return self.ans